### ISSUE TITLE
Surprise! You got *thousands* of stale /tmp/Base* directories!

### ISSUE THREAD

**Issue by kaby76** [antlr/antlr4#3502]
I am trying to chase down why several antlr/antlr4 tests are failing on `mvn clean; mvn install` for a change I'm trying. I just noticed many `/tmp/Base*/` directories created by `mvn install` after how many builds I've done over the months. I assumed that build output files were completely under the root directory for the git clone, usually under a `target/` directory, but that's not the case. Is `/tmp/Base*/` really the best place for build output files? `mvn clean` does not clean up this stuff and it was a complete surprise to find all this junk.

**Comment by ericvergnaud** [comment]
does your OS not purge tmp regularly in the background ?

**Comment by kaby76** [comment]
@ericvergnaud Probably not. This particular file system is on a WSL2 Ubuntu image and I hadn't done much to configure it. I haven't tried Maven over in a Windows cmd.exe environment to see where these files get placed, but if it's c:/temp/, that never gets emptied out. I also use Cygwin and Msys2 (both), and /tmp there doesn't get cleaned out there ever because it's not an OS.

**Comment by ericvergnaud** [comment]
https://www.fosslinux.com/41739/linux-tmp-directory-everything-you-need-to-know.htm <https://www.fosslinux.com/41739/linux-tmp-directory-everything-you-need-to-know.htm>

also works on Mac

temp is a very valid place to put those files imho
(not saying we shouldn’t clean up if required)

**Comment by kaby76** [comment]
Yeah, I disagree. I would prefer it somewhere under the clone directory because I could then just do a "rm -rf antlr4/" and be done with the git clone and all the build files. It's simple, easy to remember, doesn't require me to do "mvn clean; cd ..; rm -rf antlr4/" or some new method to devise, and works under the virtual or host machine. But, it is what it is.

**Comment by jcking** [comment]
I don't believe maven itself is creating those temp files. I think that is the antlr4 test harness for each runtime testsuite which maven is running. Ideally the antlr4 test harness for the runtime testsuite should just cleanup after.

**Comment by kaby76** [comment]
@jcking Yes, that's exactly when they are created. Each `mvn install` creates about a thousand of these Base*/ directories. The directories do come in handy after the build when evaluating what went wrong with a change. Just wish I could tell where to put them even if /tmp has to be the default. I have two different PRs that I'm working on and `mvn install` intermingles both in /tmp.

**Comment by ericvergnaud** [comment]
I believe this is already fixed since each successful test automatically deletes its test folder ?
